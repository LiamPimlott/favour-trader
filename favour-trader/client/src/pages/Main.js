import React, {Component} from 'react';
import ReviewSkillsModal from '../components/ReviewSkillsModal.js';
import axios from 'axios'
import {Row, Col} from 'antd';
import MatchCard from "../components/MatchCard";

class Main extends Component {
    constructor() {
        super();
        this.state = {
            matchedUsers: [],
            selectedUser: {},
            modalOpen: false,
        };
        this.toggleModal = this.toggleModal.bind(this);
        this.renderMatches = this.renderMatches.bind(this);
    }

    componentDidMount() {
        const { authService } = this.props;
		
        if (authService.loggedIn()) {
            const config = {
                headers: {
                    Authorization: authService.getToken()
                },
				params: {
					has: 'true',
					wants: 'true'
				}
            };

            axios.get('/api/users/matches', config)
            .then(res => res.data.matches)
            .then(matches => this.setState({matchedUsers: matches }))
            .catch( (err) => {
                console.log(err);
            });
        }
        // else should redirect to login page
    }

    toggleModal(user) {
        this.setState({
            selectedUser: user,
            modalOpen: !this.state.modalOpen,
        });
    }

    renderMatches(matches) {
        return (
            <div style={{textAlign: "center"}}>
                {
                    (matches) ? (
                        <div className={'container'}>
                            <h1>Matched Traders</h1>
                            <Row gutter={16}>
                                {
                                    matches.map(function (match) {
                                        return (
                                            <Col key={match._id} xs={24} sm={24} md={12} lg={8}>
                                                <MatchCard 
                                                    key={match._id}
                                                    user={match}
                                                    reveal={this.toggleModal.bind(this, match)}
                                                />
                                            </Col>
                                        )
                                    }, this)
                                }
                            </Row>
                        </div>
                    ) : (
                        ''
                    )
                }
            </div>
        );
    }

    render() {
        return (
            <div>
                <div className={'center-helper'}>
                    { this.state.matchedUsers ?
                        this.renderMatches(this.state.matchedUsers) :
                        "Sorry, No Matches :( Try updating the skills you are seeking."
                    }
                </div>
                <ReviewSkillsModal isOpen={this.state.modalOpen} toggle={this.toggleModal.bind(this, {})}
                               user={this.state.selectedUser}/>
            </div>
        );
    }
}

export default Main;