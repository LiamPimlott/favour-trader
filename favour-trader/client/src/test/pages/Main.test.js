import React from 'react';
import renderer from 'react-test-renderer';
import Main from '../../pages/Main.js';
import { mount, configure } from 'enzyme';
import AuthServiceMock from '../components/AuthServiceMock.js';
import moxios from "moxios";
import Adapter from "enzyme-adapter-react-16/build/index";

var authServiceMock, wrapper;

beforeEach(function () {
    configure({ adapter: new Adapter() });
    moxios.install();
    authServiceMock = new AuthServiceMock();
    wrapper = mount(<Main authService={authServiceMock} />);
});

afterEach(function () {
    moxios.uninstall();
    authServiceMock = null;
    wrapper = null;
});


it('renders correctly', () => {
  var authServiceMock = new AuthServiceMock();
  const tree = renderer
    .create(<Main authService={authServiceMock}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('verify Main Page', function () {
    expect(wrapper.contains([<h1>Matched Traders</h1>])).toEqual(true);
});