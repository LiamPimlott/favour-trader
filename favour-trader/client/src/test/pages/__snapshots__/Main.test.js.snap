// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="center-helper"
  >
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <div
        className="container"
      >
        <h1>
          Matched Traders
        </h1>
        <div
          className="ant-row"
          style={
            Object {
              "marginLeft": -8,
              "marginRight": -8,
            }
          }
        />
      </div>
    </div>
  </div>
  <div>
    
  </div>
</div>
`;
